### ISSUE TITLE
how to build

### ISSUE THREAD

**Issue by zboboj** [robidev/seapath-test-tools#1]
hello ,I want to know how to build this project. when I was building this project ,I was face a lot of problem,I can not build success,thank you.

root@xxx:~/seapath-test-tools# make
[ 12%] Building C object kernel/pkt-mirror/CMakeFiles/kern-pkt-mirror.dir/kern-pkt-mirror.c.o
[ 25%] Linking C executable kern-pkt-mirror
[ 25%] Built target kern-pkt-mirror
[ 37%] Building C object kernel/smv9-2-publisher/CMakeFiles/kern-smv9-2-publisher.dir/kern-smv9-2-publisher.c.o
[ 50%] Linking C executable kern-smv9-2-publisher
[ 50%] Built target kern-smv9-2-publisher
[ 62%] Building C object dpdk/pkt-mirror/CMakeFiles/dpdk-pkt-mirror.dir/dpdk-pkt-mirror.c.o
/root/seapath-test-tools/dpdk/pkt-mirror/dpdk-pkt-mirror.c:7:10: fatal error: rte_eal.h: No such file or directory
 #include <rte_eal.h>
          ^~~~~~~~~~~
compilation terminated.
dpdk/pkt-mirror/CMakeFiles/dpdk-pkt-mirror.dir/build.make:62: recipe for target 'dpdk/pkt-mirror/CMakeFiles/dpdk-pkt-mirror.dir/dpdk-pkt-mirror.c.o' failed
make[2]: *** [dpdk/pkt-mirror/CMakeFiles/dpdk-pkt-mirror.dir/dpdk-pkt-mirror.c.o] Error 1
CMakeFiles/Makefile2:237: recipe for target 'dpdk/pkt-mirror/CMakeFiles/dpdk-pkt-mirror.dir/all' failed
make[1]: *** [dpdk/pkt-mirror/CMakeFiles/dpdk-pkt-mirror.dir/all] Error 2
Makefile:140: recipe for target 'all' failed
make: *** [all] Error 2

root@xxx:~/seapath-test-tools# ls /usr/lib/x86_64-linux-gnu/libm.so
/usr/lib/x86_64-linux-gnu/libm.so
root@xxx:~/seapath-test-tools# make 
[ 12%] Building C object kernel/pkt-mirror/CMakeFiles/kern-pkt-mirror.dir/kern-pkt-mirror.c.o
[ 25%] Linking C executable kern-pkt-mirror
[ 25%] Built target kern-pkt-mirror
[ 37%] Building C object kernel/smv9-2-publisher/CMakeFiles/kern-smv9-2-publisher.dir/kern-smv9-2-publisher.c.o
[ 50%] Linking C executable kern-smv9-2-publisher
CMakeFiles/kern-smv9-2-publisher.dir/kern-smv9-2-publisher.c.o: In function main':
kern-smv9-2-publisher.c:(.text+0x57e): undefined reference to sin'
collect2: error: ld returned 1 exit status
kernel/smv9-2-publisher/CMakeFiles/kern-smv9-2-publisher.dir/build.make:94: recipe for target 'kernel/smv9-2-publisher/kern-smv9-2-publisher' failed
make[2]: *** [kernel/smv9-2-publisher/kern-smv9-2-publisher] Error 1
CMakeFiles/Makefile2:161: recipe for target 'kernel/smv9-2-publisher/CMakeFiles/kern-smv9-2-publisher.dir/all' failed
make[1]: *** [kernel/smv9-2-publisher/CMakeFiles/kern-smv9-2-publisher.dir/all] Error 2
Makefile:140: recipe for target 'all' failed
make: *** [all] Error 2
root@xxx:~/seapath-test-tools#

**Comment by robidev** [comment]
Hello,

Nice you are interested in this project, but perhaps you can explain what
you are trying to do. This repo is part of a larger project;

https://github.com/seapath, and requires seapath(a yocto image build of a
complete linux distro) to be build with meta-seapath-tools. I used these
small applications to test the performance of the kernel and dpdk drivers
within seapath.

The test ssuite and report can be seen here:

https://github.com/robidev/moongen-rfc2544

In short the build fails because the project is not stand-alone, and
expects some dpdk and seapath libraries to be present.

If you can explain your goal, I might be able to assist in reaching it.

Best regards,

Robin



Op di 5 nov 2024 15:17 schreef zboboj ***@***.***>:

**Comment by zboboj** [comment]
Can you leave your persion email ? I will  explain,thank you

**Comment by robidev** [comment]
Sure, robin.dev[at]gmail.com

Op wo 6 nov 2024 02:20 schreef zboboj ***@***.***>:
